WordPress/WordPressTest/AccountServiceTests.swift

```diff
@@ -210,15 +210,21 @@ class AccountServiceTests: CoreDataTestCase {
         account1.addBlogs(createMockBlogs(withIDs: [1, 2, 3, 4, 5, 6], in: context))
         account2.addBlogs(createMockBlogs(withIDs: [1, 2, 3], in: context))
         account3.addBlogs(createMockBlogs(withIDs: [4, 5, 6], in: context))
-        contextManager.save(context)
+        contextManager.saveContextAndWait(context)
         accountService.setDefaultWordPressComAccount(account1)
+        try XCTAssertEqual(context.count(for: WPAccount.fetchRequest()), 3)
 
         accountService.mergeDuplicatesIfNecessary()
-        contextManager.save(context)
+        contextManager.saveContextAndWait(context)
+        try XCTAssertEqual(context.count(for: WPAccount.fetchRequest()), 2)
+
+        let accountIds = try context.fetch(WPAccount.fetchRequest()).compactMap { ($0 as? WPAccount)?.uuid }
+        XCTAssertEqual(Set(accountIds), Set([account1.uuid, account3.uuid]))
+    }
 
-        XCTAssertFalse(account1.isDeleted)
-        XCTAssertTrue(account2.isDeleted)
-        XCTAssertFalse(account3.isDeleted)
+    func createAccount(withUsername username: String, authToken: String) -> WPAccount {
+        let id = accountService.createOrUpdateAccount(withUsername: username, authToken: authToken)
+        return try! mainContext.existingObject(with: id) as! WPAccount
```

Comment:
I realize this is just a test helper, but do we want to crash here, or have this helper `throws`?

Reply:
Addressed in https://github.com/wordpress-mobile/WordPress-iOS/pull/19893/commits/0d5b64030dbbedb7b8cc6179c838c812452fc0ec